lib/vfscore/main.c

```diff
@@ -1318,6 +1318,59 @@ UK_SYSCALL_R_DEFINE(int, rename, const char*, oldpath, const char*, newpath)
 	return -error;
 }
 
+UK_TRACEPOINT(trace_vfs_renameat, "\"%s\" \"%s\"", const char*, const char*);
+UK_TRACEPOINT(trace_vfs_renameat_ret, "");
+UK_TRACEPOINT(trace_vfs_renameat_err, "%d", int);
+
+UK_SYSCALL_R_DEFINE(int, renameat, int, olddirfd, const char*, oldpath, int, newdirfd, const char*, newpath) {
+	char src[PATH_MAX];
+	char dest[PATH_MAX];
+
+	if (!(oldpath[0] == '/' || olddirfd == AT_FDCWD)) {
+		struct vfscore_file *fp;
+		int error = fget(olddirfd, &fp);
+		if (error)
+				return -error;
+
+		struct vnode *vp = fp->f_dentry->d_vnode;
+		vn_lock(vp);
+
+		/* build absolute path */
+		strlcpy(src, fp->f_dentry->d_mount->m_path, PATH_MAX);
+		strlcat(src, fp->f_dentry->d_path, PATH_MAX);
+		strlcat(src, "/", PATH_MAX);
+		strlcat(src, oldpath, PATH_MAX);
```

Comment:
Good ol' Schlemiel at it again :smile: (if you're not familiar, check out [this article](https://www.joelonsoftware.com/2001/12/11/back-to-basics/) for a fun read).
Unfortunately much of vfscore is ripe with this; IMO not worth fixing now since a replacement is on the way, but I wanted to draw attention to it.